plugins/modules/monitoring/newrelic_deployment_v2.py

```diff
@@ -0,0 +1,158 @@
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1', 'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = \
+    '''
+---
+module: newrelic_deployment_v2
+version_added: "1.2.0"
+author: "Davinder Pal (@116davinder)"
+short_description: Notify newrelic about app deployments using newrelic v2 api
+description:
+   - Notify newrelic about app deployments
+        (https://docs.newrelic.com/docs/apm/new-relic-apm/maintenance/record-deployments)
+options:
+  token:
+    description:
+      - API token, to place in the x-api-key header.
+    required: true
+    type: str
+  app_name:
+    description:
+      - (one of app_name or application_id is required)
+        The value of app_name in the newrelic.yml file used by the application
+    required: false
+    type: str
+  application_id:
+    description:
+      - (one of app_name or application_id is required)
+        (see https://rpm.newrelic.com/api/explore/applications/list)
+    required: false
+    type: str
+  changelog:
+    description:
+      - A list of changes for this deployment
+    required: false
+    type: str
+  description:
+    description:
+      - Text annotation for the deployment - notes for you
+    required: false
+    type: str
+  revision:
+    description:
+      - A revision number (e.g., git commit SHA)
+    required: true
+    type: str
+  user:
+    description:
+      - The name of the user/process that triggered this deployment
+    required: false
+    type: str
+  validate_certs:
+    description:
+      - If C(no), SSL certificates will not be validated.
+        This should only be used for personal self-signed certificates.
+    required: false
+    default: 'yes'
+    type: bool
+'''
+
+EXAMPLES = \
+    '''
+- newrelic_deployment_v2:
+    token: XXXXXXXXX
+    app_name: ansible_app
+    user: ansible_deployment_user
+    revision: '1.X'
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import fetch_url
+import json
+
+
+# ===========================================
+# Module execution.
+#
+
+def main():
+
+    module = AnsibleModule(
+        argument_spec=dict(
+            token=dict(required=True, no_log=True),
+            app_name=dict(required=False),
+            application_id=dict(required=False),
+            changelog=dict(required=False),
+            description=dict(required=False),
+            revision=dict(required=True),
+            user=dict(required=False),
+            validate_certs=dict(default='True', type='bool'),
+        ),
+        required_one_of=[['app_name', 'application_id']]
+    )
+
+    if module.params['app_name'] and module.params['application_id']:
+        module.fail_json(msg="both app_name' and 'application_id'\
+        are defined")
+
+    if module.params['app_name']:
+        data = 'filter[name]=' + str(module.params['app_name'])
+        newrelic_api = 'https://api.newrelic.com/v2/applications.json'
```

Comment:
Why don't you filter by app name, to avoid a too long (or even paginated!) results list? (https://rpm.newrelic.com/api/explore/applications/list)

If the filter is just comparing substrings, you need to handle pagination though (since there can be more than 1 result).

Reply:
FYI: **there is nothing called `appname` in v2 API at least.** 

I am doing filter by `app_name`, which should be unique for each cluster/set of application 3 instances.
**filter[name] = app_name or Filter by application name** ( https://rpm.newrelic.com/api/explore/applications/list )

If there are more results then I am blindly trusting NewRelic Search API to give me best matched result on TOP and that will be used. 
My Code to Handle that
https://github.com/ansible-collections/community.general/blob/59d3e914fd675eda81abcd5460bbb9a33647f1bd/plugins/modules/monitoring/newrelic_deployment_v2.py#L119

**Ref:**
1. https://docs.newrelic.com/docs/agents/manage-apm-agents/app-naming/name-your-application